Tests/Services/NfieldSurveySampleTests.cs

```diff
@@ -527,6 +525,74 @@ public void TestClearAsync_ParamsAreOk_Successful()
 
         #endregion
 
+        #region UpdateAsync
+
+        [Fact]
+        public void TestUpdateAsync_SurveyIdIsNull_Throws()
+        {
+            var target = new NfieldSurveySampleService();
+
+            Assert.Throws<ArgumentNullException>(() =>
+                UnwrapAggregateException(target.UpdateAsync(null, 0, null)));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_SurveyIdIsEmpty_Throws()
+        {
+            var target = new NfieldSurveySampleService();
+
+            Assert.Throws<ArgumentException>(() =>
+                UnwrapAggregateException(target.UpdateAsync(string.Empty, 0, null)));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_SurveyIdIsWhitespace_Throws()
+        {
+            var target = new NfieldSurveySampleService();
+
+            Assert.Throws<ArgumentException>(() =>
+                UnwrapAggregateException(target.UpdateAsync("  ", 0, null)));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_ColumnsAreNull_Throws()
+        {
+            var target = new NfieldSurveySampleService();
+
+            Assert.Throws<ArgumentNullException>(() =>
+                UnwrapAggregateException(target.UpdateAsync(SurveyId, 0, null)));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_ColumnsAreEmpty_Throws()
+        {
+            var target = new NfieldSurveySampleService();
+
+            Assert.Throws<ArgumentException>(() =>
+                UnwrapAggregateException(target.UpdateAsync(SurveyId, 0, new List<SampleColumnUpdate>())));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_ParamsAreOk_Successful()
+        {
+            int sampleRecordId = 1;
+            var columns = new List<SampleColumnUpdate>();
+            columns.Add(new SampleColumnUpdate { ColumnName = "dummyCol", Value = "dummyVal" });
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+
+            mockedHttpClient.Setup(client => client.PutAsJsonAsync($"{ServiceAddress}Surveys/{SurveyId}/Sample/Update",
+                It.Is<SurveyUpdateSampleRecordModel>(c => string.IsNullOrEmpty(SurveyId) && c.ColumnUpdates.Any(n => n.ColumnName == "dummyCol"))))
```

Comment:
The SurveyId condition is unnecessary, instead, you should add the sampleRecordId to the condition, as it's used to build the SurveyUpdateSampleRecordModel

Reply:
fixed in commit 61b186c5103693627e2922122aa481b00ae36a20